libyul/optimiser/UnusedStoreBase.cpp

```diff
@@ -37,47 +37,50 @@ void UnusedStoreBase::operator()(If const& _if)
 {
 	visit(*_if.condition);
 
-	TrackedStores skipBranch{m_stores};
+	ActiveStores skipBranch{m_activeStores};
 	(*this)(_if.body);
 
-	merge(m_stores, std::move(skipBranch));
+	merge(m_activeStores, std::move(skipBranch));
 }
 
 void UnusedStoreBase::operator()(Switch const& _switch)
 {
 	visit(*_switch.expression);
 
-	TrackedStores const preState{m_stores};
+	ActiveStores const preState{m_activeStores};
 
 	bool hasDefault = false;
-	vector<TrackedStores> branches;
+	vector<ActiveStores> branches;
 	for (auto const& c: _switch.cases)
 	{
 		if (!c.value)
 			hasDefault = true;
 		(*this)(c.body);
-		branches.emplace_back(std::move(m_stores));
-		m_stores = preState;
+		branches.emplace_back(std::move(m_activeStores));
+		m_activeStores = preState;
 	}
 
 	if (hasDefault)
 	{
-		m_stores = std::move(branches.back());
+		m_activeStores = std::move(branches.back());
 		branches.pop_back();
 	}
 	for (auto& branch: branches)
-		merge(m_stores, std::move(branch));
+		merge(m_activeStores, std::move(branch));
 }
 
 void UnusedStoreBase::operator()(FunctionDefinition const& _functionDefinition)
 {
-	ScopedSaveAndRestore outerAssignments(m_stores, {});
+	ScopedSaveAndRestore allStores(m_allStores, {});
+	ScopedSaveAndRestore usedStoresStores(m_usedStores, {});
```

Comment:
```suggestion
	ScopedSaveAndRestore usedStores(m_usedStores, {});
```
?
Or `usedStoresRestore`, but `usedStoresStores` seems a bit odd.